### PR TITLE
INT-3197: Docs to AsciiDoc from DocBook

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/INT-3197

Initial Review Only

ePub needs work; we need to do side-by-side comparison of PDF and HTML.

### CODE REVIEW

**Comment by garyrussell** [comment]
Pushed some polishing.

**Comment by garyrussell** [comment]
Pushed changes for issues found by side-by-side comparison of htmlsingle.

TODO: PDF comparison, ePub decision (fix format or don't publish at this time).

**Comment by artembilan** [comment]
Is this OK?
> :asciidoctorasciidoctor: WARNING: system-management.adoc: line -1: no callouts refer to list item 5

File: build.gradle [hunk 1]
```diff
@@ -689,12 +692,40 @@ project("spring-integration-bom") {
...
+		]
+	]
+}
+
+apply plugin: DocbookReferencePlugin
 
 reference {
-	sourceDir = file('src/reference/docbook')
+	sourceFileName = 'index.xml'
+	sourceDir = file("$buildDir/html")
```
Comment (artembilan) [inline build.gradle]: `sourceDir = asciidoctor.outputDir`
to minimize target value usage.

**Comment by garyrussell** [comment]
I have had trouble tracking down the source of that warning but I will try again...

**Comment by artembilan** [comment]
Comments regarding PDF result:
1. Is that OK that Copyright paragraph has been moved now to the separate page? 
2. "Preface" header has been missed
3. Note has different style
4. `TcpConnectionServerExceptionEvent`s issue in "What's new". Similar to that we had in AMQP. Move "s" to the brackets?
5. And existing sentence "See Table 29.7, “TCP Outbound Gateway Attributes” for more information". The new one - "See ??? for more information.". Wrong link ?
6. `<inbound-channel-adapter/>`s issue
7. "Figure 2.1. Message" tip for the `Message` image. I won't mind for the tips, but seems for me they should be centered
8. `Service Activator` section. The `A request-reply "Service Activator" endpoint connects` is very small, as well as the next two
9. `2.5 Configuration` - bad code style for the sentence: "a MessageHandler` @Bean` as well as a ConsumerEndpointFactoryBean` @Bean`."
10. `Tip` block is without an icon, too.
11. `Important` - the same issue about icon
12. There is a `Note` with "The load-balancer and failover options" and it code block is outside of the `Note` paragraph. However I think we can live with that.
13. `Channel Interceptor Configuration`. Code block doesn't understand `emphasis` decoration.
14. `Global Channel Interceptor Configuration`. "or" between code blocks is a code block too :smile: 
15. `3.2 Poller (Polling Consumer)` - org.springframework.integration.core.SubscribableChannel -> org.springframework.messaging.SubscribableChannel. The same for `PollableChannel`.
16. `Configuring An Inbound Channel Adapter`. Missed `Important` label for the `Poller Configuration`. But from here it looks like we can put code block in such a paragraph. It's for the the item 12 above.

**Comment by garyrussell** [comment]
1. Yes it's ok
2. I noticed that - it looks like we have a choice of "double" preface titles in HTML or missing in PDF - but I'll look into it again
3. I noticed that too but looked at the SF PDF and it has that style note (also 10 and 11)

I will address the other bugs

**Comment by artembilan** [comment]
For me tables look good in the new style. I'd say even better with word wraps :smile:

**Comment by garyrussell** [comment]
I figured out the warning - the error was in logging-channel-adapter.adoc - misleading message.

**Comment by artembilan** [comment]
The `The MessageBuilder Helper Class` section. Code block after "MessageHeaders also defines constants for the pre-defined header names" isn't Java.

**Comment by artembilan** [comment]
`RecipientListRouterManagement`. `@ManagedResource` isn't like a code and quotes are there.

**Comment by artembilan** [comment]
`AggregatingMessageHandler` section: dangling `[source,java]` before `AbstractCorrelatingMessageHandler` ctor definition.

**Comment by artembilan** [comment]
`Caution` has been renamed to `Warning` :smile:

**Comment by artembilan** [comment]
Looks like Asciidoc has some issue with code rendering. 
`Configuring a Scatter-Gather Endpoint`: "Here, we configure the RecipientListRouter``distributor bean"

**Comment by artembilan** [comment]
`Transformers and Spring Expression Language (SpEL)`: we really should point out there the link to the latest SF Manual. However it may be done in the separate JIRA

**Comment by artembilan** [comment]
The same for the link to SpEL in the `Header Enricher` section.

**Comment by artembilan** [comment]
`Payload Enricher`, the item # 6: looks like ASciidoc can't render correctly those paragraphs. It is itemized for possible expressions in the docbook variant.

**Comment by artembilan** [comment]
`7.1 Message Endpoints`: Looks like one whitespace has been eaten before the link:

> The main reason is that there are two types of consumers:http://www.eaipatterns.com/
PollingConsumer.html[Polling Consumers] and Event Driven Consumers.

**Comment by artembilan** [comment]
BTW. The latest master has some additions to docs. Hence this should be rebased and asciidoc reconverted

**Comment by artembilan** [comment]
`Service Activators and the Spring Expression Language (SpEL)`: the same SpEL link issue.

**Comment by garyrussell** [comment]
I pushed a fix for your initial comments (and a lot of similar rendering problems).

Still TODO: small images and your latest comments

Also it looks like fixing the image caption alignment needs a new asciidoctor and/or stylesheet:

https://github.com/asciidoctor/asciidoctor/issues/857

**Comment by artembilan** [comment]
JMX chapter. `Time-Based Average Estimates` has bad render for two math functions code blocks

**Comment by artembilan** [comment]
In the same JMX: `Consolidated MBeans` has bad render for the `objectName`s. Looks like they should be represent as code block, not inline

**Comment by garyrussell** [comment]
Another small push

**Comment by garyrussell** [comment]
Pushed

**Comment by garyrussell** [comment]
And another

**Comment by garyrussell** [comment]
Squashed; ported changes since conversion; removed docbook

File: src/reference/asciidoc/polling-consumer.adoc [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+[source, java]
+Message<?> afterReceive(Message<?> result, MessageSource<?> source)
+
+This method is called after the `receive()` method; again, you can reconfigure the source, or take any action perhaps depending on the result (which can be `null` if there was no message created by the source).
+You can even return a different message!
+
+===== SimpleActiveIdleMessageSourceAdvice
+
+This advice is a simple implementation of `AbstractMessageSourceAdvice`, when used in conjunction with a `DynamicPeriodicTrigger`, it adjuststhe polling frequency depending on whether or not the previous poll resulted in a message or not.
+The poller must also have a reference to the same `DynamicPeriodicTrigger`.
```
Comment (garyrussell) [inline src/reference/asciidoc/polling-consumer.adoc]: Space needed `adjusts the`.

Please fix during merge.

**Comment by artembilan** [comment]
Merged after rebase and squesh

**Pull request closed without merging**